packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +635,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+      oldChangeEnd = oldTextLength;
+    } else {
+      for (let i = 1; i < newTextLength && oldTextLength - i >= changeStart; i++) {
+        newChangeEnd = newTextLength - i - 1;
+        oldChangeEnd = oldTextLength - i - 1;
+
+        if (newText[newChangeEnd] !== oldText[oldChangeEnd]) {
+          // Change is found
+          break;
+        }
+      }
+      // make indexes exclusive
+      newChangeEnd += 1;
+      oldChangeEnd += 1;
+    }
+  } else if (oldTextLength > newTextLength) {
+    //deletion or replacement
+    if (newTextLength === changeStart) {
+      // when change was at the end of string
+      // Change is found
+      newChangeEnd = newTextLength;
+      oldChangeEnd = oldTextLength;
+    } else {
+      for (let i = 1; i < oldTextLength && newTextLength - i >= changeStart; i++) {
+        newChangeEnd = newTextLength - i - 1;
+        oldChangeEnd = oldTextLength - i - 1;
+        if (newText[newChangeEnd] !== oldText[oldChangeEnd]) {
+          // Change is found
+          break;
+        }
+      }
+      // make indexes exclusive
+      newChangeEnd += 1;
+      oldChangeEnd += 1;
+    }
+  } else {
+    //replacement
+    for (let i = 1; i < oldTextLength && oldTextLength - i >= changeStart; i++) {
+      newChangeEnd = newTextLength - i - 1;
+      oldChangeEnd = oldTextLength - i - 1;
+
+      if (newText[newChangeEnd] !== oldText[oldChangeEnd]) {
+        // Change is found
+        break;
+      }
+    }
+    // make indexes exclusive if they aren't equal to the length of the string
+    if (newChangeEnd !== newText.length) {
+      newChangeEnd += 1;
+    }
+    if (oldChangeEnd !== oldText.length) {
+      oldChangeEnd += 1;
+    }
+  }
+  return { changeStart, oldChangeEnd, newChangeEnd };
+};
+
+/* @conditional-compile-remove(mention) */
+const htmlStringForMentionSuggestion = (suggestion: Mention, localeStrings: ComponentStrings): string => {
+  const idHTML = ' id ="' + suggestion.id + '"';
+  const displayTextHTML = ' displayText ="' + suggestion.displayText + '"';
+  const displayText = getDisplayNameForMentionSuggestion(suggestion, localeStrings);
+  return '<msft-mention' + idHTML + displayTextHTML + '>' + displayText + '</msft-mention>';
+};
+
+/* @conditional-compile-remove(mention) */
+const getDisplayNameForMentionSuggestion = (suggestion: Mention, localeStrings: ComponentStrings): string => {
+  const displayNamePlaceholder = localeStrings.participantItem.displayNamePlaceholder;
+  return suggestion.displayText !== '' ? suggestion.displayText : displayNamePlaceholder ?? '';
+};
+
+/* @conditional-compile-remove(mention) */
+type TagData = {
+  tagType: string; // The type of tag (e.g. msft-mention)
+  openTagIdx: number; // Start of the tag relative to the parent content
+  openTagBody: string; // Complete open tag body
+  content?: string; // All content between the open and close tags
+  closeTagIdx?: number; // Start of the close tag relative to the parent content
```

Comment:
closeTagPosition? Index feels like something for array

Reply:
We can revisit in the upcoming PRs.